communication/src/TlsWriteQueue.h

```diff
@@ -36,14 +36,16 @@ static constexpr size_t MAX_QUEUE_SIZE_IN_BYTES = 1024 * 1024 * 1024;  // 1 GB
 static constexpr size_t MSG_HEADER_SIZE = 4;
 
 struct OutgoingMsg {
-  OutgoingMsg(const char* raw_msg, const size_t len)
-      : msg(len + MSG_HEADER_SIZE), send_time(std::chrono::steady_clock::now()) {
-    uint32_t msg_size = htonl(static_cast<uint32_t>(len));
+  OutgoingMsg(std::vector<uint8_t>&& raw_msg)
+      : msg(raw_msg.size() + MSG_HEADER_SIZE), send_time(std::chrono::steady_clock::now()) {
+    uint32_t msg_size = htonl(static_cast<uint32_t>(raw_msg.size()));
     std::memcpy(msg.data(), &msg_size, MSG_HEADER_SIZE);
-    std::memcpy(msg.data() + MSG_HEADER_SIZE, raw_msg, len);
+    std::memcpy(msg.data() + MSG_HEADER_SIZE, raw_msg.data(), raw_msg.size());
```

Comment:
Just a note here for posterity, and because it's mentioned in the commit message as well: We can get rid of this allocation and copy by using a [ConstBufferSequence](https://think-async.com/Asio/boost_asio_1_18_1/doc/html/boost_asio/reference/ConstBufferSequence.html) in asio. This is similar to the `writev` syscall and allows writing consecutively from multiple buffers. This should be done in a follow up PR though. No need to do it now.